contracts/RevenueDistributionToken.sol

```diff
@@ -274,8 +274,6 @@ contract RevenueDistributionToken is IRevenueDistributionToken, ERC20 {
 
         require(callerAllowance >= shares_, "RDT:CALLER_ALLOWANCE");
 
-        if (callerAllowance == type(uint256).max) return;
-
         allowance[owner_][caller_] = callerAllowance - shares_;
     }
```

Comment:
We should actually just remove this function entirely and use the _approve function

Reply:
we should use decreaseAllowance

Reply:
> we should use decreaseAllowance

Nevermind its external, we should use `_approve(msg.sender, spender_, allowance[msg.sender][spender_] - subtractedAmount_);`